{
  "cells": [
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "0aStgWSO0E0E"
      },
      "source": [
        "# **PREDICT SALE PRICE**"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "1eLEkw5O0ECa"
      },
      "source": [
        "## Objectives\n",
        "\n",
        "* Fit and evaluate a regression model to predict the **Sale Price** of a house.\n",
        "\n",
        "## Inputs\n",
        "\n",
        "* outputs/datasets/collection/house_prices_records.csv\n",
        "* instructions on which variables to use for data cleaning and feature engineering. They are found on its respectives notebooks.\n",
        "\n",
        "## Outputs\n",
        "\n",
        "* Train set (features and target)\n",
        "* Test set (features and target)\n",
        "* Data cleaning and Feature Engineering pipeline\n",
        "* Modeling pipeline\n",
        "* Features importance plot\n",
        "\n",
        "## Additional Comments\n",
        "\n",
        "In this Notebook we will follow four step to manage achieve our objective:\n",
        "* Step 1: Load Data\n",
        "* Step 2: ML Pipeline with all data\n",
        "* Step 3: Refit pipeline with best features\n",
        "* Step 4: Push files to Repo\n"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "9uWZXH9LwoQg"
      },
      "source": [
        "---"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "cqP-UeN-z3i2"
      },
      "source": [
        "# Change working directory"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "aOGIGS-uz3i2"
      },
      "source": [
        "We need to change the working directory from its current folder to its parent folder\n",
        "* We access the current directory with os.getcwd()"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 3,
      "metadata": {
        "id": "wZfF_j-Bz3i4",
        "outputId": "66943449-1436-4c3d-85c7-b85f9f78349b"
      },
      "outputs": [
        {
          "data": {
            "text/plain": [
              "'/workspace/heritage-housing-issues/jupyter_notebooks'"
            ]
          },
          "execution_count": 3,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "import os\n",
        "current_dir = os.getcwd()\n",
        "current_dir"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "9MWW8E7lz3i7"
      },
      "source": [
        "We want to make the parent of the current directory the new current directory\n",
        "* os.path.dirname() gets the parent directory\n",
        "* os.chir() defines the new current directory"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 4,
      "metadata": {
        "id": "TwHsQRWjz3i9",
        "outputId": "86849db3-cd2f-4cc5-ebb8-2d0caafa1a2c"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "You set a new current directory\n"
          ]
        }
      ],
      "source": [
        "os.chdir(os.path.dirname(current_dir))\n",
        "print(\"You set a new current directory\")"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "M_xPk_Ijz3i-"
      },
      "source": [
        "Confirm the new current directory"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 5,
      "metadata": {
        "id": "vz3S-_kjz3jA",
        "outputId": "00b79ae4-75d0-4a96-d193-ac9ef9847ea2"
      },
      "outputs": [
        {
          "data": {
            "text/plain": [
              "'/workspace/heritage-housing-issues'"
            ]
          },
          "execution_count": 5,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "current_dir = os.getcwd()\n",
        "current_dir"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "---"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "-mavJ8DibrcQ"
      },
      "source": [
        "# Step 1: Load Data"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 6,
      "metadata": {},
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "(1460, 24)\n"
          ]
        },
        {
          "data": {
            "text/html": [
              "<div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>1stFlrSF</th>\n",
              "      <th>2ndFlrSF</th>\n",
              "      <th>BedroomAbvGr</th>\n",
              "      <th>BsmtExposure</th>\n",
              "      <th>BsmtFinSF1</th>\n",
              "      <th>BsmtFinType1</th>\n",
              "      <th>BsmtUnfSF</th>\n",
              "      <th>EnclosedPorch</th>\n",
              "      <th>GarageArea</th>\n",
              "      <th>GarageFinish</th>\n",
              "      <th>...</th>\n",
              "      <th>LotFrontage</th>\n",
              "      <th>MasVnrArea</th>\n",
              "      <th>OpenPorchSF</th>\n",
              "      <th>OverallCond</th>\n",
              "      <th>OverallQual</th>\n",
              "      <th>TotalBsmtSF</th>\n",
              "      <th>WoodDeckSF</th>\n",
              "      <th>YearBuilt</th>\n",
              "      <th>YearRemodAdd</th>\n",
              "      <th>SalePrice</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>856</td>\n",
              "      <td>854.0</td>\n",
              "      <td>3.0</td>\n",
              "      <td>No</td>\n",
              "      <td>706</td>\n",
              "      <td>GLQ</td>\n",
              "      <td>150</td>\n",
              "      <td>0.0</td>\n",
              "      <td>548</td>\n",
              "      <td>RFn</td>\n",
              "      <td>...</td>\n",
              "      <td>65.0</td>\n",
              "      <td>196.0</td>\n",
              "      <td>61</td>\n",
              "      <td>5</td>\n",
              "      <td>7</td>\n",
              "      <td>856</td>\n",
              "      <td>0.0</td>\n",
              "      <td>2003</td>\n",
              "      <td>2003</td>\n",
              "      <td>208500</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>1262</td>\n",
              "      <td>0.0</td>\n",
              "      <td>3.0</td>\n",
              "      <td>Gd</td>\n",
              "      <td>978</td>\n",
              "      <td>ALQ</td>\n",
              "      <td>284</td>\n",
              "      <td>NaN</td>\n",
              "      <td>460</td>\n",
              "      <td>RFn</td>\n",
              "      <td>...</td>\n",
              "      <td>80.0</td>\n",
              "      <td>0.0</td>\n",
              "      <td>0</td>\n",
              "      <td>8</td>\n",
              "      <td>6</td>\n",
              "      <td>1262</td>\n",
              "      <td>NaN</td>\n",
              "      <td>1976</td>\n",
              "      <td>1976</td>\n",
              "      <td>181500</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>920</td>\n",
              "      <td>866.0</td>\n",
              "      <td>3.0</td>\n",
              "      <td>Mn</td>\n",
              "      <td>486</td>\n",
              "      <td>GLQ</td>\n",
              "      <td>434</td>\n",
              "      <td>0.0</td>\n",
              "      <td>608</td>\n",
              "      <td>RFn</td>\n",
              "      <td>...</td>\n",
              "      <td>68.0</td>\n",
              "      <td>162.0</td>\n",
              "      <td>42</td>\n",
              "      <td>5</td>\n",
              "      <td>7</td>\n",
              "      <td>920</td>\n",
              "      <td>NaN</td>\n",
              "      <td>2001</td>\n",
              "      <td>2002</td>\n",
              "      <td>223500</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>961</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>No</td>\n",
              "      <td>216</td>\n",
              "      <td>ALQ</td>\n",
              "      <td>540</td>\n",
              "      <td>NaN</td>\n",
              "      <td>642</td>\n",
              "      <td>Unf</td>\n",
              "      <td>...</td>\n",
              "      <td>60.0</td>\n",
              "      <td>0.0</td>\n",
              "      <td>35</td>\n",
              "      <td>5</td>\n",
              "      <td>7</td>\n",
              "      <td>756</td>\n",
              "      <td>NaN</td>\n",
              "      <td>1915</td>\n",
              "      <td>1970</td>\n",
              "      <td>140000</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>1145</td>\n",
              "      <td>NaN</td>\n",
              "      <td>4.0</td>\n",
              "      <td>Av</td>\n",
              "      <td>655</td>\n",
              "      <td>GLQ</td>\n",
              "      <td>490</td>\n",
              "      <td>0.0</td>\n",
              "      <td>836</td>\n",
              "      <td>RFn</td>\n",
              "      <td>...</td>\n",
              "      <td>84.0</td>\n",
              "      <td>350.0</td>\n",
              "      <td>84</td>\n",
              "      <td>5</td>\n",
              "      <td>8</td>\n",
              "      <td>1145</td>\n",
              "      <td>NaN</td>\n",
              "      <td>2000</td>\n",
              "      <td>2000</td>\n",
              "      <td>250000</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "<p>5 rows × 24 columns</p>\n",
              "</div>"
            ],
            "text/plain": [
              "   1stFlrSF  2ndFlrSF  BedroomAbvGr BsmtExposure  BsmtFinSF1 BsmtFinType1  \\\n",
              "0       856     854.0           3.0           No         706          GLQ   \n",
              "1      1262       0.0           3.0           Gd         978          ALQ   \n",
              "2       920     866.0           3.0           Mn         486          GLQ   \n",
              "3       961       NaN           NaN           No         216          ALQ   \n",
              "4      1145       NaN           4.0           Av         655          GLQ   \n",
              "\n",
              "   BsmtUnfSF  EnclosedPorch  GarageArea GarageFinish  ...  LotFrontage  \\\n",
              "0        150            0.0         548          RFn  ...         65.0   \n",
              "1        284            NaN         460          RFn  ...         80.0   \n",
              "2        434            0.0         608          RFn  ...         68.0   \n",
              "3        540            NaN         642          Unf  ...         60.0   \n",
              "4        490            0.0         836          RFn  ...         84.0   \n",
              "\n",
              "   MasVnrArea OpenPorchSF  OverallCond  OverallQual  TotalBsmtSF  WoodDeckSF  \\\n",
              "0       196.0          61            5            7          856         0.0   \n",
              "1         0.0           0            8            6         1262         NaN   \n",
              "2       162.0          42            5            7          920         NaN   \n",
              "3         0.0          35            5            7          756         NaN   \n",
              "4       350.0          84            5            8         1145         NaN   \n",
              "\n",
              "   YearBuilt  YearRemodAdd  SalePrice  \n",
              "0       2003          2003     208500  \n",
              "1       1976          1976     181500  \n",
              "2       2001          2002     223500  \n",
              "3       1915          1970     140000  \n",
              "4       2000          2000     250000  \n",
              "\n",
              "[5 rows x 24 columns]"
            ]
          },
          "execution_count": 6,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "import numpy as np\n",
        "import pandas as pd\n",
        "df = (pd.read_csv(\"outputs/datasets/collection/house_prices_records.csv\")\n",
        "    )\n",
        "\n",
        "print(df.shape)\n",
        "df.head()"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 7,
      "metadata": {},
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "<class 'pandas.core.frame.DataFrame'>\n",
            "RangeIndex: 1460 entries, 0 to 1459\n",
            "Data columns (total 24 columns):\n",
            " #   Column         Non-Null Count  Dtype  \n",
            "---  ------         --------------  -----  \n",
            " 0   1stFlrSF       1460 non-null   int64  \n",
            " 1   2ndFlrSF       1374 non-null   float64\n",
            " 2   BedroomAbvGr   1361 non-null   float64\n",
            " 3   BsmtExposure   1460 non-null   object \n",
            " 4   BsmtFinSF1     1460 non-null   int64  \n",
            " 5   BsmtFinType1   1346 non-null   object \n",
            " 6   BsmtUnfSF      1460 non-null   int64  \n",
            " 7   EnclosedPorch  136 non-null    float64\n",
            " 8   GarageArea     1460 non-null   int64  \n",
            " 9   GarageFinish   1298 non-null   object \n",
            " 10  GarageYrBlt    1379 non-null   float64\n",
            " 11  GrLivArea      1460 non-null   int64  \n",
            " 12  KitchenQual    1460 non-null   object \n",
            " 13  LotArea        1460 non-null   int64  \n",
            " 14  LotFrontage    1201 non-null   float64\n",
            " 15  MasVnrArea     1452 non-null   float64\n",
            " 16  OpenPorchSF    1460 non-null   int64  \n",
            " 17  OverallCond    1460 non-null   int64  \n",
            " 18  OverallQual    1460 non-null   int64  \n",
            " 19  TotalBsmtSF    1460 non-null   int64  \n",
            " 20  WoodDeckSF     155 non-null    float64\n",
            " 21  YearBuilt      1460 non-null   int64  \n",
            " 22  YearRemodAdd   1460 non-null   int64  \n",
            " 23  SalePrice      1460 non-null   int64  \n",
            "dtypes: float64(7), int64(13), object(4)\n",
            "memory usage: 273.9+ KB\n"
          ]
        }
      ],
      "source": []
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {},
      "outputs": [],
      "source": [
        "df.info()"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "ZY3l0-AxO93d"
      },
      "source": [
        "---"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "uFQo3ycuO-v6"
      },
      "source": [
        "# Step 2: ML Pipeline with all data"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "### Cleaning and Feature Engineering"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 5,
      "metadata": {},
      "outputs": [],
      "source": [
        "from sklearn.pipeline import Pipeline\n",
        "\n",
        "### Data Cleaning\n",
        "from sklearn.pipeline import Pipeline\n",
        "from feature_engine.imputation import CategoricalImputer\n",
        "from feature_engine.imputation import MeanMedianImputer\n",
        "from feature_engine.imputation import ArbitraryNumberImputer\n",
        "\n",
        "### Feature Engineering\n",
        "from feature_engine.selection import SmartCorrelatedSelection\n",
        "from feature_engine.encoding import OrdinalEncoder\n",
        "from feature_engine.outliers import Winsorizer\n",
        "from feature_engine import transformation as vt\n",
        "\n",
        "### Feat Scaling\n",
        "from sklearn.preprocessing import StandardScaler\n",
        "\n",
        "### Feat Selection\n",
        "from sklearn.feature_selection import SelectFromModel\n",
        "\n",
        "### ML algorithms \n",
        "from sklearn.tree import DecisionTreeRegressor\n",
        "from xgboost import XGBRegressor\n",
        "from sklearn.ensemble import GradientBoostingRegressor, RandomForestRegressor\n",
        "from sklearn.linear_model import LinearRegression\n",
        "from sklearn.ensemble import AdaBoostRegressor\n",
        "from sklearn.ensemble import ExtraTreesRegressor\n",
        "\n",
        "def PipelineClf(model):\n",
        "  pipeline_base = Pipeline([\n",
        "    # Data Cleaning\n",
        "    (\"CategoricalImputer\",CategoricalImputer(imputation_method='missing',fill_value='Unf',\n",
        "                                                variables=['GarageFinish', 'BsmtFinType1'])),\n",
        "    \n",
        "    (\"ArbitraryNumberImputer\",ArbitraryNumberImputer(arbitrary_number=0, \n",
        "                                                variables=['2ndFlrSF', 'EnclosedPorch', 'MasVnrArea', 'WoodDeckSF'])),\n",
        "    \n",
        "    (\"MeanMedianImputer\",MeanMedianImputer(imputation_method='median', \n",
        "                                                variables=['LotFrontage', 'BedroomAbvGr', 'GarageYrBlt'])), \n",
        "     # Feature Engineer\n",
        "     (\"OrdinalCategoricalEncoder\",OrdinalEncoder(encoding_method='arbitrary', \n",
        "                                                variables = ['BsmtExposure', 'BsmtFinType1', 'GarageFinish', 'KitchenQual'])),\n",
        "\n",
        "    (\"LogTransformer\",vt.LogTransformer(variables = ['1stFlrSF', 'GrLivArea', 'LotArea', 'LotFrontage' ])),\n",
        "\n",
        "    (\"PowerrTransformer\",vt.PowerTransformer(variables = ['2ndFlrSF', 'BsmtFinSF1', 'BsmtUnfSF', 'GarageArea', \n",
        "                                                            'MasVnrArea', 'OpenPorchSF','TotalBsmtSF', 'WoodDeckSF'])),\n",
        "\n",
        "    (\"Winsorizer\",Winsorizer(capping_method='iqr', tail='both', fold=1.5, \n",
        "                                      variables=['GrLivArea'])),                                                    \n",
        "\n",
        "    (\"SmartCorrelatedSelection\",SmartCorrelatedSelection(variables=None, method=\"spearman\", \n",
        "                                                        threshold=0.8, selection_method=\"variance\")),\n",
        "    (\"scaler\",StandardScaler() ),\n",
        "\n",
        "    (\"feat_selection\",SelectFromModel(model) ),\n",
        "\n",
        "    (\"model\",model ),  \n",
        "  ])\n",
        "\n",
        "  return pipeline_base"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "### ML Pipeline for Modelling and Hyperparameter Optimization"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "Custom Class for Hyperparameter Optimization"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 6,
      "metadata": {},
      "outputs": [],
      "source": [
        "from sklearn.model_selection import GridSearchCV\n",
        "\n",
        "class HyperparameterOptimizationSearch:\n",
        "\n",
        "    def __init__(self, models, params):\n",
        "        self.models = models\n",
        "        self.params = params\n",
        "        self.keys = models.keys()\n",
        "        self.grid_searches = {}\n",
        "\n",
        "    def fit(self, X, y, cv, n_jobs, verbose=1, scoring=None, refit=False):\n",
        "        for key in self.keys:\n",
        "            print(f\"\\nRunning GridSearchCV for {key} \\n\")\n",
        "\n",
        "            model =  PipelineClf(self.models[key])\n",
        "            params = self.params[key]\n",
        "            gs = GridSearchCV(model, params, cv=cv, n_jobs=n_jobs, verbose=verbose, scoring=scoring, )\n",
        "            gs.fit(X,y)\n",
        "            self.grid_searches[key] = gs    \n",
        "\n",
        "    def score_summary(self, sort_by='mean_score'):\n",
        "        def row(key, scores, params):\n",
        "            d = {\n",
        "                 'estimator': key,\n",
        "                 'min_score': min(scores),\n",
        "                 'max_score': max(scores),\n",
        "                 'mean_score': np.mean(scores),\n",
        "                 'std_score': np.std(scores),\n",
        "            }\n",
        "            return pd.Series({**params,**d})\n",
        "\n",
        "        rows = []\n",
        "        for k in self.grid_searches:\n",
        "            params = self.grid_searches[k].cv_results_['params']\n",
        "            scores = []\n",
        "            for i in range(self.grid_searches[k].cv):\n",
        "                key = \"split{}_test_score\".format(i)\n",
        "                r = self.grid_searches[k].cv_results_[key]        \n",
        "                scores.append(r.reshape(len(params),1))\n",
        "\n",
        "            all_scores = np.hstack(scores)\n",
        "            for p, s in zip(params,all_scores):\n",
        "                rows.append((row(k, s, p)))\n",
        "\n",
        "        df = pd.concat(rows, axis=1).T.sort_values([sort_by], ascending=False)\n",
        "        columns = ['estimator', 'min_score', 'mean_score', 'max_score', 'std_score']\n",
        "        columns = columns + [c for c in df.columns if c not in columns]\n",
        "        return df[columns], self.grid_searches"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "### Split Train and Test Set"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 7,
      "metadata": {},
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "* Train set: (1168, 23) (1168,) \n",
            "* Test set: (292, 23) (292,)\n"
          ]
        }
      ],
      "source": [
        "from sklearn.model_selection import train_test_split\n",
        "X_train, X_test,y_train, y_test = train_test_split(\n",
        "                                    df.drop(['SalePrice'],axis=1),\n",
        "                                    df['SalePrice'],\n",
        "                                    test_size = 0.2,\n",
        "                                    random_state = 0,\n",
        "                                    )\n",
        "\n",
        "print(\"* Train set:\", X_train.shape, y_train.shape, \"\\n* Test set:\",  X_test.shape, y_test.shape)"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "### Grid Search CV - Sklearn\n",
        "\n",
        "#### Use default hyperparameters to find most suitable algorithm"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 8,
      "metadata": {},
      "outputs": [],
      "source": [
        "models_quick_search = {\n",
        "    'LinearRegression': LinearRegression(),\n",
        "    \"DecisionTreeRegressor\": DecisionTreeRegressor(random_state=0),\n",
        "    \"RandomForestRegressor\": RandomForestRegressor(random_state=0),\n",
        "    \"ExtraTreesRegressor\": ExtraTreesRegressor(random_state=0),\n",
        "    \"AdaBoostRegressor\": AdaBoostRegressor(random_state=0),\n",
        "    \"GradientBoostingRegressor\": GradientBoostingRegressor(random_state=0),\n",
        "    \"XGBRegressor\": XGBRegressor(random_state=0),\n",
        "}\n",
        "\n",
        "params_quick_search = {\n",
        "    'LinearRegression': {},\n",
        "    \"DecisionTreeRegressor\": {},\n",
        "    \"RandomForestRegressor\": {},\n",
        "    \"ExtraTreesRegressor\": {},\n",
        "    \"AdaBoostRegressor\": {},\n",
        "    \"GradientBoostingRegressor\": {},\n",
        "    \"XGBRegressor\": {},\n",
        "}"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "Run A GridSearch CV - Binary Classifier"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 9,
      "metadata": {},
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "\n",
            "Running GridSearchCV for LinearRegression \n",
            "\n",
            "Fitting 5 folds for each of 1 candidates, totalling 5 fits\n"
          ]
        },
        {
          "name": "stderr",
          "output_type": "stream",
          "text": [
            "/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/model_selection/_validation.py:696: UserWarning: Scoring failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
            "Traceback (most recent call last):\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 687, in _score\n",
            "    scores = scorer(estimator, X_test, y_test)\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_scorer.py\", line 199, in __call__\n",
            "    return self._score(partial(_cached_call, None), estimator, X, y_true,\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_scorer.py\", line 242, in _score\n",
            "    return self._sign * self._score_func(y_true, y_pred,\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 63, in inner_f\n",
            "    return f(*args, **kwargs)\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_classification.py\", line 1774, in recall_score\n",
            "    _, r, _, _ = precision_recall_fscore_support(y_true, y_pred,\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 63, in inner_f\n",
            "    return f(*args, **kwargs)\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_classification.py\", line 1464, in precision_recall_fscore_support\n",
            "    labels = _check_set_wise_labels(y_true, y_pred, average, labels,\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_classification.py\", line 1277, in _check_set_wise_labels\n",
            "    y_type, y_true, y_pred = _check_targets(y_true, y_pred)\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_classification.py\", line 92, in _check_targets\n",
            "    raise ValueError(\"Classification metrics can't handle a mix of {0} \"\n",
            "ValueError: Classification metrics can't handle a mix of multiclass and continuous targets\n",
            "\n",
            "  warnings.warn(\n",
            "/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/model_selection/_validation.py:696: UserWarning: Scoring failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
            "Traceback (most recent call last):\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 687, in _score\n",
            "    scores = scorer(estimator, X_test, y_test)\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_scorer.py\", line 199, in __call__\n",
            "    return self._score(partial(_cached_call, None), estimator, X, y_true,\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_scorer.py\", line 242, in _score\n",
            "    return self._sign * self._score_func(y_true, y_pred,\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 63, in inner_f\n",
            "    return f(*args, **kwargs)\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_classification.py\", line 1774, in recall_score\n",
            "    _, r, _, _ = precision_recall_fscore_support(y_true, y_pred,\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 63, in inner_f\n",
            "    return f(*args, **kwargs)\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_classification.py\", line 1464, in precision_recall_fscore_support\n",
            "    labels = _check_set_wise_labels(y_true, y_pred, average, labels,\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_classification.py\", line 1277, in _check_set_wise_labels\n",
            "    y_type, y_true, y_pred = _check_targets(y_true, y_pred)\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_classification.py\", line 92, in _check_targets\n",
            "    raise ValueError(\"Classification metrics can't handle a mix of {0} \"\n",
            "ValueError: Classification metrics can't handle a mix of multiclass and continuous targets\n",
            "\n",
            "  warnings.warn(\n",
            "/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/model_selection/_validation.py:696: UserWarning: Scoring failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
            "Traceback (most recent call last):\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 687, in _score\n",
            "    scores = scorer(estimator, X_test, y_test)\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_scorer.py\", line 199, in __call__\n",
            "    return self._score(partial(_cached_call, None), estimator, X, y_true,\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_scorer.py\", line 242, in _score\n",
            "    return self._sign * self._score_func(y_true, y_pred,\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 63, in inner_f\n",
            "    return f(*args, **kwargs)\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_classification.py\", line 1774, in recall_score\n",
            "    _, r, _, _ = precision_recall_fscore_support(y_true, y_pred,\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 63, in inner_f\n",
            "    return f(*args, **kwargs)\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_classification.py\", line 1464, in precision_recall_fscore_support\n",
            "    labels = _check_set_wise_labels(y_true, y_pred, average, labels,\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_classification.py\", line 1277, in _check_set_wise_labels\n",
            "    y_type, y_true, y_pred = _check_targets(y_true, y_pred)\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_classification.py\", line 92, in _check_targets\n",
            "    raise ValueError(\"Classification metrics can't handle a mix of {0} \"\n",
            "ValueError: Classification metrics can't handle a mix of multiclass and continuous targets\n",
            "\n",
            "  warnings.warn(\n",
            "/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/model_selection/_validation.py:696: UserWarning: Scoring failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
            "Traceback (most recent call last):\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 687, in _score\n",
            "    scores = scorer(estimator, X_test, y_test)\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_scorer.py\", line 199, in __call__\n",
            "    return self._score(partial(_cached_call, None), estimator, X, y_true,\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_scorer.py\", line 242, in _score\n",
            "    return self._sign * self._score_func(y_true, y_pred,\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 63, in inner_f\n",
            "    return f(*args, **kwargs)\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_classification.py\", line 1774, in recall_score\n",
            "    _, r, _, _ = precision_recall_fscore_support(y_true, y_pred,\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 63, in inner_f\n",
            "    return f(*args, **kwargs)\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_classification.py\", line 1464, in precision_recall_fscore_support\n",
            "    labels = _check_set_wise_labels(y_true, y_pred, average, labels,\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_classification.py\", line 1277, in _check_set_wise_labels\n",
            "    y_type, y_true, y_pred = _check_targets(y_true, y_pred)\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_classification.py\", line 92, in _check_targets\n",
            "    raise ValueError(\"Classification metrics can't handle a mix of {0} \"\n",
            "ValueError: Classification metrics can't handle a mix of multiclass and continuous targets\n",
            "\n",
            "  warnings.warn(\n",
            "/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/model_selection/_validation.py:696: UserWarning: Scoring failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
            "Traceback (most recent call last):\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 687, in _score\n",
            "    scores = scorer(estimator, X_test, y_test)\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_scorer.py\", line 199, in __call__\n",
            "    return self._score(partial(_cached_call, None), estimator, X, y_true,\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_scorer.py\", line 242, in _score\n",
            "    return self._sign * self._score_func(y_true, y_pred,\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 63, in inner_f\n",
            "    return f(*args, **kwargs)\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_classification.py\", line 1774, in recall_score\n",
            "    _, r, _, _ = precision_recall_fscore_support(y_true, y_pred,\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 63, in inner_f\n",
            "    return f(*args, **kwargs)\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_classification.py\", line 1464, in precision_recall_fscore_support\n",
            "    labels = _check_set_wise_labels(y_true, y_pred, average, labels,\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_classification.py\", line 1277, in _check_set_wise_labels\n",
            "    y_type, y_true, y_pred = _check_targets(y_true, y_pred)\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_classification.py\", line 92, in _check_targets\n",
            "    raise ValueError(\"Classification metrics can't handle a mix of {0} \"\n",
            "ValueError: Classification metrics can't handle a mix of multiclass and continuous targets\n",
            "\n",
            "  warnings.warn(\n",
            "/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/model_selection/_search.py:922: UserWarning: One or more of the test scores are non-finite: [nan]\n",
            "  warnings.warn(\n"
          ]
        },
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "\n",
            "Running GridSearchCV for DecisionTreeRegressor \n",
            "\n",
            "Fitting 5 folds for each of 1 candidates, totalling 5 fits\n"
          ]
        },
        {
          "name": "stderr",
          "output_type": "stream",
          "text": [
            "/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/model_selection/_validation.py:696: UserWarning: Scoring failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
            "Traceback (most recent call last):\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 687, in _score\n",
            "    scores = scorer(estimator, X_test, y_test)\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_scorer.py\", line 199, in __call__\n",
            "    return self._score(partial(_cached_call, None), estimator, X, y_true,\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_scorer.py\", line 242, in _score\n",
            "    return self._sign * self._score_func(y_true, y_pred,\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 63, in inner_f\n",
            "    return f(*args, **kwargs)\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_classification.py\", line 1774, in recall_score\n",
            "    _, r, _, _ = precision_recall_fscore_support(y_true, y_pred,\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 63, in inner_f\n",
            "    return f(*args, **kwargs)\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_classification.py\", line 1464, in precision_recall_fscore_support\n",
            "    labels = _check_set_wise_labels(y_true, y_pred, average, labels,\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_classification.py\", line 1277, in _check_set_wise_labels\n",
            "    y_type, y_true, y_pred = _check_targets(y_true, y_pred)\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_classification.py\", line 92, in _check_targets\n",
            "    raise ValueError(\"Classification metrics can't handle a mix of {0} \"\n",
            "ValueError: Classification metrics can't handle a mix of multiclass and continuous targets\n",
            "\n",
            "  warnings.warn(\n",
            "/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/model_selection/_validation.py:696: UserWarning: Scoring failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
            "Traceback (most recent call last):\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 687, in _score\n",
            "    scores = scorer(estimator, X_test, y_test)\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_scorer.py\", line 199, in __call__\n",
            "    return self._score(partial(_cached_call, None), estimator, X, y_true,\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_scorer.py\", line 242, in _score\n",
            "    return self._sign * self._score_func(y_true, y_pred,\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 63, in inner_f\n",
            "    return f(*args, **kwargs)\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_classification.py\", line 1774, in recall_score\n",
            "    _, r, _, _ = precision_recall_fscore_support(y_true, y_pred,\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 63, in inner_f\n",
            "    return f(*args, **kwargs)\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_classification.py\", line 1464, in precision_recall_fscore_support\n",
            "    labels = _check_set_wise_labels(y_true, y_pred, average, labels,\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_classification.py\", line 1277, in _check_set_wise_labels\n",
            "    y_type, y_true, y_pred = _check_targets(y_true, y_pred)\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_classification.py\", line 92, in _check_targets\n",
            "    raise ValueError(\"Classification metrics can't handle a mix of {0} \"\n",
            "ValueError: Classification metrics can't handle a mix of multiclass and continuous targets\n",
            "\n",
            "  warnings.warn(\n",
            "/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/model_selection/_validation.py:696: UserWarning: Scoring failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
            "Traceback (most recent call last):\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 687, in _score\n",
            "    scores = scorer(estimator, X_test, y_test)\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_scorer.py\", line 199, in __call__\n",
            "    return self._score(partial(_cached_call, None), estimator, X, y_true,\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_scorer.py\", line 242, in _score\n",
            "    return self._sign * self._score_func(y_true, y_pred,\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 63, in inner_f\n",
            "    return f(*args, **kwargs)\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_classification.py\", line 1774, in recall_score\n",
            "    _, r, _, _ = precision_recall_fscore_support(y_true, y_pred,\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 63, in inner_f\n",
            "    return f(*args, **kwargs)\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_classification.py\", line 1464, in precision_recall_fscore_support\n",
            "    labels = _check_set_wise_labels(y_true, y_pred, average, labels,\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_classification.py\", line 1277, in _check_set_wise_labels\n",
            "    y_type, y_true, y_pred = _check_targets(y_true, y_pred)\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_classification.py\", line 92, in _check_targets\n",
            "    raise ValueError(\"Classification metrics can't handle a mix of {0} \"\n",
            "ValueError: Classification metrics can't handle a mix of multiclass and continuous targets\n",
            "\n",
            "  warnings.warn(\n",
            "/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/model_selection/_validation.py:696: UserWarning: Scoring failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
            "Traceback (most recent call last):\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 687, in _score\n",
            "    scores = scorer(estimator, X_test, y_test)\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_scorer.py\", line 199, in __call__\n",
            "    return self._score(partial(_cached_call, None), estimator, X, y_true,\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_scorer.py\", line 242, in _score\n",
            "    return self._sign * self._score_func(y_true, y_pred,\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 63, in inner_f\n",
            "    return f(*args, **kwargs)\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_classification.py\", line 1774, in recall_score\n",
            "    _, r, _, _ = precision_recall_fscore_support(y_true, y_pred,\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 63, in inner_f\n",
            "    return f(*args, **kwargs)\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_classification.py\", line 1464, in precision_recall_fscore_support\n",
            "    labels = _check_set_wise_labels(y_true, y_pred, average, labels,\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_classification.py\", line 1277, in _check_set_wise_labels\n",
            "    y_type, y_true, y_pred = _check_targets(y_true, y_pred)\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_classification.py\", line 92, in _check_targets\n",
            "    raise ValueError(\"Classification metrics can't handle a mix of {0} \"\n",
            "ValueError: Classification metrics can't handle a mix of multiclass and continuous targets\n",
            "\n",
            "  warnings.warn(\n",
            "/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/model_selection/_validation.py:696: UserWarning: Scoring failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
            "Traceback (most recent call last):\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 687, in _score\n",
            "    scores = scorer(estimator, X_test, y_test)\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_scorer.py\", line 199, in __call__\n",
            "    return self._score(partial(_cached_call, None), estimator, X, y_true,\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_scorer.py\", line 242, in _score\n",
            "    return self._sign * self._score_func(y_true, y_pred,\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 63, in inner_f\n",
            "    return f(*args, **kwargs)\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_classification.py\", line 1774, in recall_score\n",
            "    _, r, _, _ = precision_recall_fscore_support(y_true, y_pred,\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 63, in inner_f\n",
            "    return f(*args, **kwargs)\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_classification.py\", line 1464, in precision_recall_fscore_support\n",
            "    labels = _check_set_wise_labels(y_true, y_pred, average, labels,\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_classification.py\", line 1277, in _check_set_wise_labels\n",
            "    y_type, y_true, y_pred = _check_targets(y_true, y_pred)\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_classification.py\", line 92, in _check_targets\n",
            "    raise ValueError(\"Classification metrics can't handle a mix of {0} \"\n",
            "ValueError: Classification metrics can't handle a mix of multiclass and continuous targets\n",
            "\n",
            "  warnings.warn(\n",
            "/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/model_selection/_search.py:922: UserWarning: One or more of the test scores are non-finite: [nan]\n",
            "  warnings.warn(\n"
          ]
        },
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "\n",
            "Running GridSearchCV for RandomForestRegressor \n",
            "\n",
            "Fitting 5 folds for each of 1 candidates, totalling 5 fits\n"
          ]
        },
        {
          "name": "stderr",
          "output_type": "stream",
          "text": [
            "/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/model_selection/_validation.py:696: UserWarning: Scoring failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
            "Traceback (most recent call last):\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 687, in _score\n",
            "    scores = scorer(estimator, X_test, y_test)\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_scorer.py\", line 199, in __call__\n",
            "    return self._score(partial(_cached_call, None), estimator, X, y_true,\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_scorer.py\", line 242, in _score\n",
            "    return self._sign * self._score_func(y_true, y_pred,\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 63, in inner_f\n",
            "    return f(*args, **kwargs)\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_classification.py\", line 1774, in recall_score\n",
            "    _, r, _, _ = precision_recall_fscore_support(y_true, y_pred,\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 63, in inner_f\n",
            "    return f(*args, **kwargs)\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_classification.py\", line 1464, in precision_recall_fscore_support\n",
            "    labels = _check_set_wise_labels(y_true, y_pred, average, labels,\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_classification.py\", line 1277, in _check_set_wise_labels\n",
            "    y_type, y_true, y_pred = _check_targets(y_true, y_pred)\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_classification.py\", line 92, in _check_targets\n",
            "    raise ValueError(\"Classification metrics can't handle a mix of {0} \"\n",
            "ValueError: Classification metrics can't handle a mix of multiclass and continuous targets\n",
            "\n",
            "  warnings.warn(\n",
            "/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/model_selection/_validation.py:696: UserWarning: Scoring failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
            "Traceback (most recent call last):\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 687, in _score\n",
            "    scores = scorer(estimator, X_test, y_test)\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_scorer.py\", line 199, in __call__\n",
            "    return self._score(partial(_cached_call, None), estimator, X, y_true,\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_scorer.py\", line 242, in _score\n",
            "    return self._sign * self._score_func(y_true, y_pred,\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 63, in inner_f\n",
            "    return f(*args, **kwargs)\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_classification.py\", line 1774, in recall_score\n",
            "    _, r, _, _ = precision_recall_fscore_support(y_true, y_pred,\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 63, in inner_f\n",
            "    return f(*args, **kwargs)\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_classification.py\", line 1464, in precision_recall_fscore_support\n",
            "    labels = _check_set_wise_labels(y_true, y_pred, average, labels,\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_classification.py\", line 1277, in _check_set_wise_labels\n",
            "    y_type, y_true, y_pred = _check_targets(y_true, y_pred)\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_classification.py\", line 92, in _check_targets\n",
            "    raise ValueError(\"Classification metrics can't handle a mix of {0} \"\n",
            "ValueError: Classification metrics can't handle a mix of multiclass and continuous targets\n",
            "\n",
            "  warnings.warn(\n",
            "/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/model_selection/_validation.py:696: UserWarning: Scoring failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
            "Traceback (most recent call last):\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 687, in _score\n",
            "    scores = scorer(estimator, X_test, y_test)\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_scorer.py\", line 199, in __call__\n",
            "    return self._score(partial(_cached_call, None), estimator, X, y_true,\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_scorer.py\", line 242, in _score\n",
            "    return self._sign * self._score_func(y_true, y_pred,\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 63, in inner_f\n",
            "    return f(*args, **kwargs)\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_classification.py\", line 1774, in recall_score\n",
            "    _, r, _, _ = precision_recall_fscore_support(y_true, y_pred,\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 63, in inner_f\n",
            "    return f(*args, **kwargs)\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_classification.py\", line 1464, in precision_recall_fscore_support\n",
            "    labels = _check_set_wise_labels(y_true, y_pred, average, labels,\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_classification.py\", line 1277, in _check_set_wise_labels\n",
            "    y_type, y_true, y_pred = _check_targets(y_true, y_pred)\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_classification.py\", line 92, in _check_targets\n",
            "    raise ValueError(\"Classification metrics can't handle a mix of {0} \"\n",
            "ValueError: Classification metrics can't handle a mix of multiclass and continuous targets\n",
            "\n",
            "  warnings.warn(\n",
            "/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/model_selection/_validation.py:696: UserWarning: Scoring failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
            "Traceback (most recent call last):\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 687, in _score\n",
            "    scores = scorer(estimator, X_test, y_test)\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_scorer.py\", line 199, in __call__\n",
            "    return self._score(partial(_cached_call, None), estimator, X, y_true,\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_scorer.py\", line 242, in _score\n",
            "    return self._sign * self._score_func(y_true, y_pred,\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 63, in inner_f\n",
            "    return f(*args, **kwargs)\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_classification.py\", line 1774, in recall_score\n",
            "    _, r, _, _ = precision_recall_fscore_support(y_true, y_pred,\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 63, in inner_f\n",
            "    return f(*args, **kwargs)\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_classification.py\", line 1464, in precision_recall_fscore_support\n",
            "    labels = _check_set_wise_labels(y_true, y_pred, average, labels,\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_classification.py\", line 1277, in _check_set_wise_labels\n",
            "    y_type, y_true, y_pred = _check_targets(y_true, y_pred)\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_classification.py\", line 92, in _check_targets\n",
            "    raise ValueError(\"Classification metrics can't handle a mix of {0} \"\n",
            "ValueError: Classification metrics can't handle a mix of multiclass and continuous targets\n",
            "\n",
            "  warnings.warn(\n",
            "/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/model_selection/_validation.py:696: UserWarning: Scoring failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
            "Traceback (most recent call last):\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 687, in _score\n",
            "    scores = scorer(estimator, X_test, y_test)\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_scorer.py\", line 199, in __call__\n",
            "    return self._score(partial(_cached_call, None), estimator, X, y_true,\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_scorer.py\", line 242, in _score\n",
            "    return self._sign * self._score_func(y_true, y_pred,\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 63, in inner_f\n",
            "    return f(*args, **kwargs)\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_classification.py\", line 1774, in recall_score\n",
            "    _, r, _, _ = precision_recall_fscore_support(y_true, y_pred,\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 63, in inner_f\n",
            "    return f(*args, **kwargs)\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_classification.py\", line 1464, in precision_recall_fscore_support\n",
            "    labels = _check_set_wise_labels(y_true, y_pred, average, labels,\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_classification.py\", line 1277, in _check_set_wise_labels\n",
            "    y_type, y_true, y_pred = _check_targets(y_true, y_pred)\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_classification.py\", line 92, in _check_targets\n",
            "    raise ValueError(\"Classification metrics can't handle a mix of {0} \"\n",
            "ValueError: Classification metrics can't handle a mix of multiclass and continuous targets\n",
            "\n",
            "  warnings.warn(\n",
            "/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/model_selection/_search.py:922: UserWarning: One or more of the test scores are non-finite: [nan]\n",
            "  warnings.warn(\n"
          ]
        },
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "\n",
            "Running GridSearchCV for ExtraTreesRegressor \n",
            "\n",
            "Fitting 5 folds for each of 1 candidates, totalling 5 fits\n"
          ]
        },
        {
          "name": "stderr",
          "output_type": "stream",
          "text": [
            "/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/model_selection/_validation.py:696: UserWarning: Scoring failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
            "Traceback (most recent call last):\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 687, in _score\n",
            "    scores = scorer(estimator, X_test, y_test)\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_scorer.py\", line 199, in __call__\n",
            "    return self._score(partial(_cached_call, None), estimator, X, y_true,\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_scorer.py\", line 242, in _score\n",
            "    return self._sign * self._score_func(y_true, y_pred,\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 63, in inner_f\n",
            "    return f(*args, **kwargs)\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_classification.py\", line 1774, in recall_score\n",
            "    _, r, _, _ = precision_recall_fscore_support(y_true, y_pred,\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 63, in inner_f\n",
            "    return f(*args, **kwargs)\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_classification.py\", line 1464, in precision_recall_fscore_support\n",
            "    labels = _check_set_wise_labels(y_true, y_pred, average, labels,\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_classification.py\", line 1277, in _check_set_wise_labels\n",
            "    y_type, y_true, y_pred = _check_targets(y_true, y_pred)\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_classification.py\", line 92, in _check_targets\n",
            "    raise ValueError(\"Classification metrics can't handle a mix of {0} \"\n",
            "ValueError: Classification metrics can't handle a mix of multiclass and continuous targets\n",
            "\n",
            "  warnings.warn(\n",
            "/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/model_selection/_validation.py:696: UserWarning: Scoring failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
            "Traceback (most recent call last):\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 687, in _score\n",
            "    scores = scorer(estimator, X_test, y_test)\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_scorer.py\", line 199, in __call__\n",
            "    return self._score(partial(_cached_call, None), estimator, X, y_true,\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_scorer.py\", line 242, in _score\n",
            "    return self._sign * self._score_func(y_true, y_pred,\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 63, in inner_f\n",
            "    return f(*args, **kwargs)\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_classification.py\", line 1774, in recall_score\n",
            "    _, r, _, _ = precision_recall_fscore_support(y_true, y_pred,\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 63, in inner_f\n",
            "    return f(*args, **kwargs)\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_classification.py\", line 1464, in precision_recall_fscore_support\n",
            "    labels = _check_set_wise_labels(y_true, y_pred, average, labels,\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_classification.py\", line 1277, in _check_set_wise_labels\n",
            "    y_type, y_true, y_pred = _check_targets(y_true, y_pred)\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_classification.py\", line 92, in _check_targets\n",
            "    raise ValueError(\"Classification metrics can't handle a mix of {0} \"\n",
            "ValueError: Classification metrics can't handle a mix of multiclass and continuous targets\n",
            "\n",
            "  warnings.warn(\n",
            "/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/model_selection/_validation.py:696: UserWarning: Scoring failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
            "Traceback (most recent call last):\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 687, in _score\n",
            "    scores = scorer(estimator, X_test, y_test)\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_scorer.py\", line 199, in __call__\n",
            "    return self._score(partial(_cached_call, None), estimator, X, y_true,\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_scorer.py\", line 242, in _score\n",
            "    return self._sign * self._score_func(y_true, y_pred,\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 63, in inner_f\n",
            "    return f(*args, **kwargs)\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_classification.py\", line 1774, in recall_score\n",
            "    _, r, _, _ = precision_recall_fscore_support(y_true, y_pred,\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 63, in inner_f\n",
            "    return f(*args, **kwargs)\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_classification.py\", line 1464, in precision_recall_fscore_support\n",
            "    labels = _check_set_wise_labels(y_true, y_pred, average, labels,\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_classification.py\", line 1277, in _check_set_wise_labels\n",
            "    y_type, y_true, y_pred = _check_targets(y_true, y_pred)\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_classification.py\", line 92, in _check_targets\n",
            "    raise ValueError(\"Classification metrics can't handle a mix of {0} \"\n",
            "ValueError: Classification metrics can't handle a mix of multiclass and continuous targets\n",
            "\n",
            "  warnings.warn(\n",
            "/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/model_selection/_validation.py:696: UserWarning: Scoring failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
            "Traceback (most recent call last):\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 687, in _score\n",
            "    scores = scorer(estimator, X_test, y_test)\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_scorer.py\", line 199, in __call__\n",
            "    return self._score(partial(_cached_call, None), estimator, X, y_true,\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_scorer.py\", line 242, in _score\n",
            "    return self._sign * self._score_func(y_true, y_pred,\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 63, in inner_f\n",
            "    return f(*args, **kwargs)\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_classification.py\", line 1774, in recall_score\n",
            "    _, r, _, _ = precision_recall_fscore_support(y_true, y_pred,\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 63, in inner_f\n",
            "    return f(*args, **kwargs)\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_classification.py\", line 1464, in precision_recall_fscore_support\n",
            "    labels = _check_set_wise_labels(y_true, y_pred, average, labels,\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_classification.py\", line 1277, in _check_set_wise_labels\n",
            "    y_type, y_true, y_pred = _check_targets(y_true, y_pred)\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_classification.py\", line 92, in _check_targets\n",
            "    raise ValueError(\"Classification metrics can't handle a mix of {0} \"\n",
            "ValueError: Classification metrics can't handle a mix of multiclass and continuous targets\n",
            "\n",
            "  warnings.warn(\n",
            "/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/model_selection/_validation.py:696: UserWarning: Scoring failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
            "Traceback (most recent call last):\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 687, in _score\n",
            "    scores = scorer(estimator, X_test, y_test)\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_scorer.py\", line 199, in __call__\n",
            "    return self._score(partial(_cached_call, None), estimator, X, y_true,\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_scorer.py\", line 242, in _score\n",
            "    return self._sign * self._score_func(y_true, y_pred,\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 63, in inner_f\n",
            "    return f(*args, **kwargs)\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_classification.py\", line 1774, in recall_score\n",
            "    _, r, _, _ = precision_recall_fscore_support(y_true, y_pred,\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 63, in inner_f\n",
            "    return f(*args, **kwargs)\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_classification.py\", line 1464, in precision_recall_fscore_support\n",
            "    labels = _check_set_wise_labels(y_true, y_pred, average, labels,\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_classification.py\", line 1277, in _check_set_wise_labels\n",
            "    y_type, y_true, y_pred = _check_targets(y_true, y_pred)\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_classification.py\", line 92, in _check_targets\n",
            "    raise ValueError(\"Classification metrics can't handle a mix of {0} \"\n",
            "ValueError: Classification metrics can't handle a mix of multiclass and continuous targets\n",
            "\n",
            "  warnings.warn(\n",
            "/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/model_selection/_search.py:922: UserWarning: One or more of the test scores are non-finite: [nan]\n",
            "  warnings.warn(\n"
          ]
        },
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "\n",
            "Running GridSearchCV for AdaBoostRegressor \n",
            "\n",
            "Fitting 5 folds for each of 1 candidates, totalling 5 fits\n"
          ]
        },
        {
          "name": "stderr",
          "output_type": "stream",
          "text": [
            "/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/model_selection/_validation.py:696: UserWarning: Scoring failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
            "Traceback (most recent call last):\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 687, in _score\n",
            "    scores = scorer(estimator, X_test, y_test)\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_scorer.py\", line 199, in __call__\n",
            "    return self._score(partial(_cached_call, None), estimator, X, y_true,\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_scorer.py\", line 242, in _score\n",
            "    return self._sign * self._score_func(y_true, y_pred,\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 63, in inner_f\n",
            "    return f(*args, **kwargs)\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_classification.py\", line 1774, in recall_score\n",
            "    _, r, _, _ = precision_recall_fscore_support(y_true, y_pred,\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 63, in inner_f\n",
            "    return f(*args, **kwargs)\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_classification.py\", line 1464, in precision_recall_fscore_support\n",
            "    labels = _check_set_wise_labels(y_true, y_pred, average, labels,\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_classification.py\", line 1277, in _check_set_wise_labels\n",
            "    y_type, y_true, y_pred = _check_targets(y_true, y_pred)\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_classification.py\", line 92, in _check_targets\n",
            "    raise ValueError(\"Classification metrics can't handle a mix of {0} \"\n",
            "ValueError: Classification metrics can't handle a mix of multiclass and continuous targets\n",
            "\n",
            "  warnings.warn(\n",
            "/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/model_selection/_validation.py:696: UserWarning: Scoring failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
            "Traceback (most recent call last):\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 687, in _score\n",
            "    scores = scorer(estimator, X_test, y_test)\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_scorer.py\", line 199, in __call__\n",
            "    return self._score(partial(_cached_call, None), estimator, X, y_true,\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_scorer.py\", line 242, in _score\n",
            "    return self._sign * self._score_func(y_true, y_pred,\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 63, in inner_f\n",
            "    return f(*args, **kwargs)\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_classification.py\", line 1774, in recall_score\n",
            "    _, r, _, _ = precision_recall_fscore_support(y_true, y_pred,\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 63, in inner_f\n",
            "    return f(*args, **kwargs)\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_classification.py\", line 1464, in precision_recall_fscore_support\n",
            "    labels = _check_set_wise_labels(y_true, y_pred, average, labels,\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_classification.py\", line 1277, in _check_set_wise_labels\n",
            "    y_type, y_true, y_pred = _check_targets(y_true, y_pred)\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_classification.py\", line 92, in _check_targets\n",
            "    raise ValueError(\"Classification metrics can't handle a mix of {0} \"\n",
            "ValueError: Classification metrics can't handle a mix of multiclass and continuous targets\n",
            "\n",
            "  warnings.warn(\n",
            "/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/model_selection/_validation.py:696: UserWarning: Scoring failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
            "Traceback (most recent call last):\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 687, in _score\n",
            "    scores = scorer(estimator, X_test, y_test)\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_scorer.py\", line 199, in __call__\n",
            "    return self._score(partial(_cached_call, None), estimator, X, y_true,\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_scorer.py\", line 242, in _score\n",
            "    return self._sign * self._score_func(y_true, y_pred,\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 63, in inner_f\n",
            "    return f(*args, **kwargs)\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_classification.py\", line 1774, in recall_score\n",
            "    _, r, _, _ = precision_recall_fscore_support(y_true, y_pred,\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 63, in inner_f\n",
            "    return f(*args, **kwargs)\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_classification.py\", line 1464, in precision_recall_fscore_support\n",
            "    labels = _check_set_wise_labels(y_true, y_pred, average, labels,\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_classification.py\", line 1277, in _check_set_wise_labels\n",
            "    y_type, y_true, y_pred = _check_targets(y_true, y_pred)\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_classification.py\", line 92, in _check_targets\n",
            "    raise ValueError(\"Classification metrics can't handle a mix of {0} \"\n",
            "ValueError: Classification metrics can't handle a mix of multiclass and continuous targets\n",
            "\n",
            "  warnings.warn(\n",
            "/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/model_selection/_validation.py:696: UserWarning: Scoring failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
            "Traceback (most recent call last):\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 687, in _score\n",
            "    scores = scorer(estimator, X_test, y_test)\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_scorer.py\", line 199, in __call__\n",
            "    return self._score(partial(_cached_call, None), estimator, X, y_true,\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_scorer.py\", line 242, in _score\n",
            "    return self._sign * self._score_func(y_true, y_pred,\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 63, in inner_f\n",
            "    return f(*args, **kwargs)\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_classification.py\", line 1774, in recall_score\n",
            "    _, r, _, _ = precision_recall_fscore_support(y_true, y_pred,\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 63, in inner_f\n",
            "    return f(*args, **kwargs)\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_classification.py\", line 1464, in precision_recall_fscore_support\n",
            "    labels = _check_set_wise_labels(y_true, y_pred, average, labels,\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_classification.py\", line 1277, in _check_set_wise_labels\n",
            "    y_type, y_true, y_pred = _check_targets(y_true, y_pred)\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_classification.py\", line 92, in _check_targets\n",
            "    raise ValueError(\"Classification metrics can't handle a mix of {0} \"\n",
            "ValueError: Classification metrics can't handle a mix of multiclass and continuous targets\n",
            "\n",
            "  warnings.warn(\n",
            "/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/model_selection/_validation.py:696: UserWarning: Scoring failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
            "Traceback (most recent call last):\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 687, in _score\n",
            "    scores = scorer(estimator, X_test, y_test)\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_scorer.py\", line 199, in __call__\n",
            "    return self._score(partial(_cached_call, None), estimator, X, y_true,\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_scorer.py\", line 242, in _score\n",
            "    return self._sign * self._score_func(y_true, y_pred,\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 63, in inner_f\n",
            "    return f(*args, **kwargs)\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_classification.py\", line 1774, in recall_score\n",
            "    _, r, _, _ = precision_recall_fscore_support(y_true, y_pred,\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 63, in inner_f\n",
            "    return f(*args, **kwargs)\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_classification.py\", line 1464, in precision_recall_fscore_support\n",
            "    labels = _check_set_wise_labels(y_true, y_pred, average, labels,\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_classification.py\", line 1277, in _check_set_wise_labels\n",
            "    y_type, y_true, y_pred = _check_targets(y_true, y_pred)\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_classification.py\", line 92, in _check_targets\n",
            "    raise ValueError(\"Classification metrics can't handle a mix of {0} \"\n",
            "ValueError: Classification metrics can't handle a mix of multiclass and continuous targets\n",
            "\n",
            "  warnings.warn(\n",
            "/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/model_selection/_search.py:922: UserWarning: One or more of the test scores are non-finite: [nan]\n",
            "  warnings.warn(\n"
          ]
        },
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "\n",
            "Running GridSearchCV for GradientBoostingRegressor \n",
            "\n",
            "Fitting 5 folds for each of 1 candidates, totalling 5 fits\n"
          ]
        },
        {
          "name": "stderr",
          "output_type": "stream",
          "text": [
            "/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/model_selection/_validation.py:696: UserWarning: Scoring failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
            "Traceback (most recent call last):\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 687, in _score\n",
            "    scores = scorer(estimator, X_test, y_test)\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_scorer.py\", line 199, in __call__\n",
            "    return self._score(partial(_cached_call, None), estimator, X, y_true,\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_scorer.py\", line 242, in _score\n",
            "    return self._sign * self._score_func(y_true, y_pred,\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 63, in inner_f\n",
            "    return f(*args, **kwargs)\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_classification.py\", line 1774, in recall_score\n",
            "    _, r, _, _ = precision_recall_fscore_support(y_true, y_pred,\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 63, in inner_f\n",
            "    return f(*args, **kwargs)\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_classification.py\", line 1464, in precision_recall_fscore_support\n",
            "    labels = _check_set_wise_labels(y_true, y_pred, average, labels,\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_classification.py\", line 1277, in _check_set_wise_labels\n",
            "    y_type, y_true, y_pred = _check_targets(y_true, y_pred)\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_classification.py\", line 92, in _check_targets\n",
            "    raise ValueError(\"Classification metrics can't handle a mix of {0} \"\n",
            "ValueError: Classification metrics can't handle a mix of multiclass and continuous targets\n",
            "\n",
            "  warnings.warn(\n",
            "/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/model_selection/_validation.py:696: UserWarning: Scoring failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
            "Traceback (most recent call last):\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 687, in _score\n",
            "    scores = scorer(estimator, X_test, y_test)\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_scorer.py\", line 199, in __call__\n",
            "    return self._score(partial(_cached_call, None), estimator, X, y_true,\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_scorer.py\", line 242, in _score\n",
            "    return self._sign * self._score_func(y_true, y_pred,\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 63, in inner_f\n",
            "    return f(*args, **kwargs)\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_classification.py\", line 1774, in recall_score\n",
            "    _, r, _, _ = precision_recall_fscore_support(y_true, y_pred,\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 63, in inner_f\n",
            "    return f(*args, **kwargs)\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_classification.py\", line 1464, in precision_recall_fscore_support\n",
            "    labels = _check_set_wise_labels(y_true, y_pred, average, labels,\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_classification.py\", line 1277, in _check_set_wise_labels\n",
            "    y_type, y_true, y_pred = _check_targets(y_true, y_pred)\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_classification.py\", line 92, in _check_targets\n",
            "    raise ValueError(\"Classification metrics can't handle a mix of {0} \"\n",
            "ValueError: Classification metrics can't handle a mix of multiclass and continuous targets\n",
            "\n",
            "  warnings.warn(\n",
            "/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/model_selection/_validation.py:696: UserWarning: Scoring failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
            "Traceback (most recent call last):\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 687, in _score\n",
            "    scores = scorer(estimator, X_test, y_test)\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_scorer.py\", line 199, in __call__\n",
            "    return self._score(partial(_cached_call, None), estimator, X, y_true,\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_scorer.py\", line 242, in _score\n",
            "    return self._sign * self._score_func(y_true, y_pred,\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 63, in inner_f\n",
            "    return f(*args, **kwargs)\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_classification.py\", line 1774, in recall_score\n",
            "    _, r, _, _ = precision_recall_fscore_support(y_true, y_pred,\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 63, in inner_f\n",
            "    return f(*args, **kwargs)\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_classification.py\", line 1464, in precision_recall_fscore_support\n",
            "    labels = _check_set_wise_labels(y_true, y_pred, average, labels,\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_classification.py\", line 1277, in _check_set_wise_labels\n",
            "    y_type, y_true, y_pred = _check_targets(y_true, y_pred)\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_classification.py\", line 92, in _check_targets\n",
            "    raise ValueError(\"Classification metrics can't handle a mix of {0} \"\n",
            "ValueError: Classification metrics can't handle a mix of multiclass and continuous targets\n",
            "\n",
            "  warnings.warn(\n",
            "/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/model_selection/_validation.py:696: UserWarning: Scoring failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
            "Traceback (most recent call last):\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 687, in _score\n",
            "    scores = scorer(estimator, X_test, y_test)\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_scorer.py\", line 199, in __call__\n",
            "    return self._score(partial(_cached_call, None), estimator, X, y_true,\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_scorer.py\", line 242, in _score\n",
            "    return self._sign * self._score_func(y_true, y_pred,\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 63, in inner_f\n",
            "    return f(*args, **kwargs)\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_classification.py\", line 1774, in recall_score\n",
            "    _, r, _, _ = precision_recall_fscore_support(y_true, y_pred,\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 63, in inner_f\n",
            "    return f(*args, **kwargs)\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_classification.py\", line 1464, in precision_recall_fscore_support\n",
            "    labels = _check_set_wise_labels(y_true, y_pred, average, labels,\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_classification.py\", line 1277, in _check_set_wise_labels\n",
            "    y_type, y_true, y_pred = _check_targets(y_true, y_pred)\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_classification.py\", line 92, in _check_targets\n",
            "    raise ValueError(\"Classification metrics can't handle a mix of {0} \"\n",
            "ValueError: Classification metrics can't handle a mix of multiclass and continuous targets\n",
            "\n",
            "  warnings.warn(\n",
            "/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/model_selection/_validation.py:696: UserWarning: Scoring failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
            "Traceback (most recent call last):\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 687, in _score\n",
            "    scores = scorer(estimator, X_test, y_test)\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_scorer.py\", line 199, in __call__\n",
            "    return self._score(partial(_cached_call, None), estimator, X, y_true,\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_scorer.py\", line 242, in _score\n",
            "    return self._sign * self._score_func(y_true, y_pred,\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 63, in inner_f\n",
            "    return f(*args, **kwargs)\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_classification.py\", line 1774, in recall_score\n",
            "    _, r, _, _ = precision_recall_fscore_support(y_true, y_pred,\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 63, in inner_f\n",
            "    return f(*args, **kwargs)\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_classification.py\", line 1464, in precision_recall_fscore_support\n",
            "    labels = _check_set_wise_labels(y_true, y_pred, average, labels,\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_classification.py\", line 1277, in _check_set_wise_labels\n",
            "    y_type, y_true, y_pred = _check_targets(y_true, y_pred)\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_classification.py\", line 92, in _check_targets\n",
            "    raise ValueError(\"Classification metrics can't handle a mix of {0} \"\n",
            "ValueError: Classification metrics can't handle a mix of multiclass and continuous targets\n",
            "\n",
            "  warnings.warn(\n",
            "/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/model_selection/_search.py:922: UserWarning: One or more of the test scores are non-finite: [nan]\n",
            "  warnings.warn(\n"
          ]
        },
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "\n",
            "Running GridSearchCV for XGBRegressor \n",
            "\n",
            "Fitting 5 folds for each of 1 candidates, totalling 5 fits\n"
          ]
        },
        {
          "name": "stderr",
          "output_type": "stream",
          "text": [
            "/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/model_selection/_validation.py:696: UserWarning: Scoring failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
            "Traceback (most recent call last):\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 687, in _score\n",
            "    scores = scorer(estimator, X_test, y_test)\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_scorer.py\", line 199, in __call__\n",
            "    return self._score(partial(_cached_call, None), estimator, X, y_true,\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_scorer.py\", line 242, in _score\n",
            "    return self._sign * self._score_func(y_true, y_pred,\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 63, in inner_f\n",
            "    return f(*args, **kwargs)\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_classification.py\", line 1774, in recall_score\n",
            "    _, r, _, _ = precision_recall_fscore_support(y_true, y_pred,\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 63, in inner_f\n",
            "    return f(*args, **kwargs)\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_classification.py\", line 1464, in precision_recall_fscore_support\n",
            "    labels = _check_set_wise_labels(y_true, y_pred, average, labels,\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_classification.py\", line 1277, in _check_set_wise_labels\n",
            "    y_type, y_true, y_pred = _check_targets(y_true, y_pred)\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_classification.py\", line 92, in _check_targets\n",
            "    raise ValueError(\"Classification metrics can't handle a mix of {0} \"\n",
            "ValueError: Classification metrics can't handle a mix of multiclass and continuous targets\n",
            "\n",
            "  warnings.warn(\n",
            "/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/model_selection/_validation.py:696: UserWarning: Scoring failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
            "Traceback (most recent call last):\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 687, in _score\n",
            "    scores = scorer(estimator, X_test, y_test)\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_scorer.py\", line 199, in __call__\n",
            "    return self._score(partial(_cached_call, None), estimator, X, y_true,\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_scorer.py\", line 242, in _score\n",
            "    return self._sign * self._score_func(y_true, y_pred,\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 63, in inner_f\n",
            "    return f(*args, **kwargs)\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_classification.py\", line 1774, in recall_score\n",
            "    _, r, _, _ = precision_recall_fscore_support(y_true, y_pred,\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 63, in inner_f\n",
            "    return f(*args, **kwargs)\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_classification.py\", line 1464, in precision_recall_fscore_support\n",
            "    labels = _check_set_wise_labels(y_true, y_pred, average, labels,\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_classification.py\", line 1277, in _check_set_wise_labels\n",
            "    y_type, y_true, y_pred = _check_targets(y_true, y_pred)\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_classification.py\", line 92, in _check_targets\n",
            "    raise ValueError(\"Classification metrics can't handle a mix of {0} \"\n",
            "ValueError: Classification metrics can't handle a mix of multiclass and continuous targets\n",
            "\n",
            "  warnings.warn(\n",
            "/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/model_selection/_validation.py:696: UserWarning: Scoring failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
            "Traceback (most recent call last):\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 687, in _score\n",
            "    scores = scorer(estimator, X_test, y_test)\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_scorer.py\", line 199, in __call__\n",
            "    return self._score(partial(_cached_call, None), estimator, X, y_true,\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_scorer.py\", line 242, in _score\n",
            "    return self._sign * self._score_func(y_true, y_pred,\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 63, in inner_f\n",
            "    return f(*args, **kwargs)\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_classification.py\", line 1774, in recall_score\n",
            "    _, r, _, _ = precision_recall_fscore_support(y_true, y_pred,\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 63, in inner_f\n",
            "    return f(*args, **kwargs)\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_classification.py\", line 1464, in precision_recall_fscore_support\n",
            "    labels = _check_set_wise_labels(y_true, y_pred, average, labels,\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_classification.py\", line 1277, in _check_set_wise_labels\n",
            "    y_type, y_true, y_pred = _check_targets(y_true, y_pred)\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_classification.py\", line 92, in _check_targets\n",
            "    raise ValueError(\"Classification metrics can't handle a mix of {0} \"\n",
            "ValueError: Classification metrics can't handle a mix of multiclass and continuous targets\n",
            "\n",
            "  warnings.warn(\n",
            "/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/model_selection/_validation.py:696: UserWarning: Scoring failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
            "Traceback (most recent call last):\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 687, in _score\n",
            "    scores = scorer(estimator, X_test, y_test)\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_scorer.py\", line 199, in __call__\n",
            "    return self._score(partial(_cached_call, None), estimator, X, y_true,\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_scorer.py\", line 242, in _score\n",
            "    return self._sign * self._score_func(y_true, y_pred,\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 63, in inner_f\n",
            "    return f(*args, **kwargs)\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_classification.py\", line 1774, in recall_score\n",
            "    _, r, _, _ = precision_recall_fscore_support(y_true, y_pred,\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 63, in inner_f\n",
            "    return f(*args, **kwargs)\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_classification.py\", line 1464, in precision_recall_fscore_support\n",
            "    labels = _check_set_wise_labels(y_true, y_pred, average, labels,\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_classification.py\", line 1277, in _check_set_wise_labels\n",
            "    y_type, y_true, y_pred = _check_targets(y_true, y_pred)\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_classification.py\", line 92, in _check_targets\n",
            "    raise ValueError(\"Classification metrics can't handle a mix of {0} \"\n",
            "ValueError: Classification metrics can't handle a mix of multiclass and continuous targets\n",
            "\n",
            "  warnings.warn(\n",
            "/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/model_selection/_validation.py:696: UserWarning: Scoring failed. The score on this train-test partition for these parameters will be set to nan. Details: \n",
            "Traceback (most recent call last):\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/model_selection/_validation.py\", line 687, in _score\n",
            "    scores = scorer(estimator, X_test, y_test)\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_scorer.py\", line 199, in __call__\n",
            "    return self._score(partial(_cached_call, None), estimator, X, y_true,\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_scorer.py\", line 242, in _score\n",
            "    return self._sign * self._score_func(y_true, y_pred,\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 63, in inner_f\n",
            "    return f(*args, **kwargs)\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_classification.py\", line 1774, in recall_score\n",
            "    _, r, _, _ = precision_recall_fscore_support(y_true, y_pred,\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/utils/validation.py\", line 63, in inner_f\n",
            "    return f(*args, **kwargs)\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_classification.py\", line 1464, in precision_recall_fscore_support\n",
            "    labels = _check_set_wise_labels(y_true, y_pred, average, labels,\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_classification.py\", line 1277, in _check_set_wise_labels\n",
            "    y_type, y_true, y_pred = _check_targets(y_true, y_pred)\n",
            "  File \"/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/metrics/_classification.py\", line 92, in _check_targets\n",
            "    raise ValueError(\"Classification metrics can't handle a mix of {0} \"\n",
            "ValueError: Classification metrics can't handle a mix of multiclass and continuous targets\n",
            "\n",
            "  warnings.warn(\n",
            "/workspace/.pip-modules/lib/python3.8/site-packages/sklearn/model_selection/_search.py:922: UserWarning: One or more of the test scores are non-finite: [nan]\n",
            "  warnings.warn(\n"
          ]
        }
      ],
      "source": [
        "from sklearn.metrics import make_scorer, recall_score\n",
        "search = HyperparameterOptimizationSearch(models=models_quick_search, params=params_quick_search)\n",
        "search.fit(X_train, y_train,\n",
        "           scoring =  make_scorer(recall_score, pos_label=1),\n",
        "           n_jobs=-1, cv=5)"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "Check results"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 10,
      "metadata": {},
      "outputs": [
        {
          "data": {
            "text/html": [
              "<div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>estimator</th>\n",
              "      <th>min_score</th>\n",
              "      <th>mean_score</th>\n",
              "      <th>max_score</th>\n",
              "      <th>std_score</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>LinearRegression</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>DecisionTreeRegressor</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>RandomForestRegressor</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>ExtraTreesRegressor</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>AdaBoostRegressor</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>5</th>\n",
              "      <td>GradientBoostingRegressor</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>6</th>\n",
              "      <td>XGBRegressor</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>"
            ],
            "text/plain": [
              "                   estimator min_score mean_score max_score std_score\n",
              "0           LinearRegression       NaN        NaN       NaN       NaN\n",
              "1      DecisionTreeRegressor       NaN        NaN       NaN       NaN\n",
              "2      RandomForestRegressor       NaN        NaN       NaN       NaN\n",
              "3        ExtraTreesRegressor       NaN        NaN       NaN       NaN\n",
              "4          AdaBoostRegressor       NaN        NaN       NaN       NaN\n",
              "5  GradientBoostingRegressor       NaN        NaN       NaN       NaN\n",
              "6               XGBRegressor       NaN        NaN       NaN       NaN"
            ]
          },
          "execution_count": 10,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "grid_search_summary, grid_search_pipelines = search.score_summary(sort_by='mean_score')\n",
        "grid_search_summary"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "---"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "# Step 3: Refit pipeline with best features"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "---"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "NOTE"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "* You may add how many sections you want, as long as it supports your project workflow.\n",
        "* All notebook's cells should be run top-down (you can't create a dynamic wherein a given point you need to go back to a previous cell to execute some task, like go back to a previous cell and refresh a variable content)"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "---"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "ltNetd085qHf"
      },
      "source": [
        "# Step 4: Push files to Repo"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "* In case you don't need to push files to Repo, you may replace this section for \"Conclusions and Next Steps\" and state your conclusions and next steps."
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "aKlnIozA4eQO",
        "outputId": "fd09bc1f-adb1-4511-f6ce-492a6af570c0"
      },
      "outputs": [],
      "source": [
        "import os\n",
        "try:\n",
        "  # create here your folder\n",
        "  # os.makedirs(name='')\n",
        "except Exception as e:\n",
        "  print(e)\n"
      ]
    }
  ],
  "metadata": {
    "accelerator": "GPU",
    "colab": {
      "name": "Data Practitioner Jupyter Notebook.ipynb",
      "provenance": [],
      "toc_visible": true
    },
    "interpreter": {
      "hash": "8b8334dab9339717f727a1deaf837b322d7a41c20d15cc86be99a8e69ceec8ce"
    },
    "kernelspec": {
      "display_name": "Python 3.8.12 64-bit ('3.8.12': pyenv)",
      "name": "python3"
    },
    "language_info": {
      "codemirror_mode": {
        "name": "ipython",
        "version": 3
      },
      "file_extension": ".py",
      "mimetype": "text/x-python",
      "name": "python",
      "nbconvert_exporter": "python",
      "pygments_lexer": "ipython3",
      "version": "3.8.12"
    },
    "orig_nbformat": 2
  },
  "nbformat": 4,
  "nbformat_minor": 2
}
